app/components/BuyQuoteBox.tsx

```diff
@@ -0,0 +1,396 @@
+import { Button } from "@nextui-org/button";
+import { Input } from "@nextui-org/input";
+import { ChangeEvent, useCallback, useEffect, useMemo, useRef, useState } from "react";
+import { Card, Link, Tooltip } from "@nextui-org/react";
+import {Select, SelectItem} from "@nextui-org/select";
+import { BuyConfigResponse, BuyOptionsRequest, BuyOptionsResponse, BuyQuoteRequest, BuyQuoteResponse} from "../utils/types";
+import { generateBuyOptions, generateBuyQuote } from "../utils/queries";
+import ReactJson from "react-json-view";
+import { BuyConfigBox } from "./BuyConfigBox";
+import SecureTokenBox from "./SecureTokenBox";
+
+
+export default function BuyQuoteBox() {
+
+    /* Formatting Variables - refs to scroll to headers, adding bottom padding conditionally */
+    const buyOptionsHeaderRef = useRef<HTMLElement | null>(null);
+    const buyQuoteHeaderRef = useRef<HTMLElement | null>(null);
+
+    useEffect(() => {
+        buyOptionsHeaderRef.current = document.getElementById('buyOptionsHeader');
+        buyQuoteHeaderRef.current = document.getElementById('buyQuoteHeader');
```

Comment:
React can set ref values for you by passing the ref as a prop to the element: 
https://react.dev/learn/manipulating-the-dom-with-refs